interfaces/repo.go

```diff
@@ -0,0 +1,569 @@
+		return r.disconnectEverythingFromPlugSlot(slotSnapName, slotName)
+	default:
+		return r.disconnectPlugFromPlugSlot(plugSnapName, plugName, slotSnapName, slotName)
+	}
+
+}
+
+// disconnectEverythingFromSnap finds a specific snap and disconnects all the plugs connected to all the slots therein.
+func (r *Repository) disconnectEverythingFromSnap(slotSnapName string) error {
+	if _, ok := r.slots[slotSnapName]; !ok {
+		return fmt.Errorf("cannot disconnect plug from snap %q, no such snap", slotSnapName)
+	}
+	for _, slot := range r.slots[slotSnapName] {
+		for plug := range r.slotPlugs[slot] {
+			r.disconnect(plug, slot)
+		}
+	}
+	return nil
+}
+
+// disconnectEverythingFromPlugSlot finds a specific plug slot and disconnects all the plugs connected there.
+func (r *Repository) disconnectEverythingFromPlugSlot(slotSnapName, slotName string) error {
+	// Ensure that such slot exists
+	slot := r.slots[slotSnapName][slotName]
+	if slot == nil {
+		return fmt.Errorf("cannot disconnect plug from slot %q from snap %q, no such slot", slotName, slotSnapName)
+	}
+	for plug := range r.slotPlugs[slot] {
+		r.disconnect(plug, slot)
+	}
+	return nil
+}
+
+// disconnectPlugFromPlugSlot finds a specific plug slot and plug and disconnects it.
+func (r *Repository) disconnectPlugFromPlugSlot(plugSnapName, plugName, slotSnapName, slotName string) error {
+	// Ensure that such plug exists
+	plug := r.plugs[plugSnapName][plugName]
+	if plug == nil {
+		return fmt.Errorf("cannot disconnect plug %q from snap %q, no such plug", plugName, plugSnapName)
+	}
+	// Ensure that such slot exists
+	slot := r.slots[slotSnapName][slotName]
+	if slot == nil {
+		return fmt.Errorf("cannot disconnect plug from slot %q from snap %q, no such slot", slotName, slotSnapName)
+	}
+	// Ensure that slot and plug are connected
+	if !r.slotPlugs[slot][plug] {
+		return fmt.Errorf("cannot disconnect plug %q from snap %q from slot %q from snap %q, it is not connected",
+			plugName, plugSnapName, slotName, slotSnapName)
+	}
+	r.disconnect(plug, slot)
+	return nil
+}
+
+// disconnect disconnects a specific plug from a specific plug slot.
+func (r *Repository) disconnect(plug *Plug, slot *Slot) {
+	delete(r.slotPlugs[slot], plug)
+	if len(r.slotPlugs[slot]) == 0 {
+		delete(r.slotPlugs, slot)
+	}
+	delete(r.plugSlots[plug], slot)
+	if len(r.plugSlots[plug]) == 0 {
+		delete(r.plugSlots, plug)
+	}
+}
+
+// ConnectedTo returns all the plugs connected to a given snap.
+func (r *Repository) ConnectedTo(snapName string) map[*Slot][]*Plug {
+	r.m.Lock()
+	defer r.m.Unlock()
+
+	result := make(map[*Slot][]*Plug)
+	for _, slot := range r.slots[snapName] {
+		for plug := range r.slotPlugs[slot] {
+			result[slot] = append(result[slot], plug)
+		}
+		sort.Sort(byPlugSnapAndName(result[slot]))
+	}
+	return result
+}
+
+// ConnectedBy returns all of the plugs connected by a given snap.
+func (r *Repository) ConnectedBy(snapName string) map[*Plug][]*Slot {
```

Comment:
ConnectedBy doesn't sound right anymore.

Let's rename the pair to:

ConnectedTo => ConnectedSlots
ConnectedBy => ConnectedPlugs